src/App.tsx

```diff
@@ -1,300 +1,100 @@
 import "./App.css";
 
 import {
-  ChangeEvent,
-  useCallback,
-  useDeferredValue,
+  useContext,
   useEffect,
   useMemo,
   useState,
-  useSyncExternalStore,
 } from "react";
 
-import { Button } from "@components/ui/button";
-import { CopyPalette } from "@components/CopyPalette.tsx";
-import { debounce } from "@utils/debounce";
-import { DeletePalette } from "@components/DeletePalette.tsx";
-import { getRandomColor } from "@utils/getRandomColor";
-import { SavePalette } from "@components/SavePalette.tsx";
-import { Save, Shuffle } from "lucide-react";
-import { toast } from "sonner";
-import { Toaster } from "@components/ui/sonner";
+import { Button } from "@/components/shared/ui/button";
+import { SavePalette } from "@/components/save-palette/SavePalette";
+import { Shuffle } from "lucide-react";
+import { Toaster } from "@/components/shared/ui/sonner";
 import { useLocation } from "wouter";
-import chroma from "chroma-js";
-import GraphicItems from "@components/GraphicItems";
-import Layout from "./layouts/Layout";
-import Palette from "@components/Palette";
-import store, { type Palettes } from "@utils/palettes";
-import LittlePalette from "./components/LittlePalette";
-import { EditPaletteName } from "./components/EditPaletteName";
-import {
-  Tooltip,
-  TooltipContent,
-  TooltipProvider,
-  TooltipTrigger,
-} from "@radix-ui/react-tooltip";
-import { Footer } from "./components/Footer";
-import PromptDialog from "@components/PromptDialog";
+import { ValidateHexadecimal } from "./utils/hexadecimal-validator";
+import { HexadecimalContext } from "./provider/hexadecimal/hexadecimal.context";
+import { Footer } from "./sections/footer/Footer";
 
-function App() {
-  const [color, setColor] = useState("#ffffff");
-  const [colorAux, setColorAux] = useState(color);
-  const deferredColor = useDeferredValue(color);
+import GraphicItems from "@components/graphic-items/GraphicItems";
+import Layout from "./layouts/Layout";
+import Palette from "@/components/palette/Palette";
+import store from "@utils/palettes";
+import ColorPicker from "./components/color-picker/ColorPicker";
+import SavePaletteSection from "./sections/save-palette-section/SavePaletteSection";
+import PromptDialog from "./components/prompt-dialog/PromptDialog";
+import AppController from "./app.controller";
 
-  const [isEditNamePalette, setIsEditNamePalette] = useState("");
-  const [valueEditNamePalette, setValueEditNamePalette] = useState("");
-  const handledEditNamePalette = (name: string) => {
-    setIsEditNamePalette(name);
-    setValueEditNamePalette(name);
-  };
 
-  const savedPalettes = useSyncExternalStore<Palettes>(
-    store.subscribe,
-    store.getSnapshot
-  );
+const controller: AppController = new AppController();
```

Comment:
(Personal opinion): having controllers for purely static functions add overhead without providing any value.

Reply:
Can you expand more on your opinion? I based it on Angular. The controller was created outside of the component so that each time the component is rebuilt, the instance is not recreated.

Also, the controller makes the function cleaner.

Reply:
Sure!

When I see a class with non-static methods, I expect those methods to either:
- Require information provided via constructor (dependency injection-like).
- Require information not available statically (class properties).

Since none of those rules apply to these functions (#1 doesn't apply since the constructor is empty, and #2 doesn't apply since the class doesn't have properties), I see these functions more fitted as static, so you don't need to instantiate the class to use them (such as `Math.random`, i.e.) and you can directly do `AppController.whatever(...)`. Such patterns kinda communicate (to me) that there're no side-effects going on inside of the class + the function only makes use of the provided parameters (and nothing else).

This is why several React architectures rely on "`namespace`s" such as
```
const AppController = {
  getThis,
  doThat,
}
```
and then consumed via `AppController.getThis(), AppController.doThat()`, instead of using classes.

Reply:
However, if someone wanted to pass a service in the future, they would need an instance of the class. Although, in this application, using static methods is a good option.